controller/src/main/java/org/apache/rocketmq/controller/impl/heartbeat/RaftBrokerHeartBeatManager.java

```diff
@@ -0,0 +1,261 @@
+
+    private final Map<Channel, BrokerIdentityInfo> brokerChannelIdentityInfoMap = new HashMap<>();
+
+    public RaftBrokerHeartBeatManager(ControllerConfig controllerConfig) {
+        this.scheduledService = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryImpl("RaftBrokerHeartbeatManager_scheduledService_"));
+        this.executor = Executors.newFixedThreadPool(2, new ThreadFactoryImpl("RaftBrokerHeartbeatManager_executorService_"));
+        this.controllerConfig = controllerConfig;
+    }
+
+    public void setController(JRaftController controller) {
+        this.controller = controller;
+    }
+
+    @Override
+    public void initialize() {
+
+    }
+
+    @Override
+    public void start() {
+        this.scheduledService.scheduleAtFixedRate(this::scanNotActiveBroker, 2000, this.controllerConfig.getScanNotActiveBrokerInterval(), TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public void shutdown() {
+        this.scheduledService.shutdown();
+        this.executor.shutdown();
+    }
+
+    @Override
+    public void registerBrokerLifecycleListener(BrokerLifecycleListener listener) {
+        brokerLifecycleListeners.add(listener);
+    }
+
+    @Override
+    public void onBrokerHeartbeat(String clusterName, String brokerName, String brokerAddr, Long brokerId,
+        Long timeoutMillis, Channel channel, Integer epoch, Long maxOffset, Long confirmOffset,
+        Integer electionPriority) {
+        BrokerIdentityInfo brokerIdentityInfo = new BrokerIdentityInfo(clusterName, brokerName, brokerId);
+        int realEpoch = Optional.ofNullable(epoch).orElse(-1);
+        long realBrokerId = Optional.ofNullable(brokerId).orElse(-1L);
+        long realMaxOffset = Optional.ofNullable(maxOffset).orElse(-1L);
+        long realConfirmOffset = Optional.ofNullable(confirmOffset).orElse(-1L);
+        long realTimeoutMillis = Optional.ofNullable(timeoutMillis).orElse(DEFAULT_BROKER_CHANNEL_EXPIRED_TIME);
+        int realElectionPriority = Optional.ofNullable(electionPriority).orElse(Integer.MAX_VALUE);
+        BrokerLiveInfo liveInfo = new BrokerLiveInfo(brokerName,
+            brokerAddr,
+            realBrokerId,
+            System.currentTimeMillis(),
+            realTimeoutMillis,
+            null,
+            realEpoch,
+            realMaxOffset,
+            realElectionPriority,
+            realConfirmOffset);
+        log.info("broker {} heart beat", brokerIdentityInfo);
+        RaftBrokerHeartBeatEventRequest requestHeader = new RaftBrokerHeartBeatEventRequest(brokerIdentityInfo, liveInfo);
+        CompletableFuture<RemotingCommand> future = controller.onBrokerHeartBeat(requestHeader);
+        try {
+            RemotingCommand remotingCommand = future.get(5, java.util.concurrent.TimeUnit.SECONDS);
+            if (remotingCommand.getCode() != ResponseCode.SUCCESS) {
+                throw new RuntimeException("on broker heartbeat return invalid code, code: " + remotingCommand.getCode());
```

Comment:
There will be a lot of error messages in follower controllers.
![image](https://github.com/apache/rocketmq/assets/21073949/6c6e2378-0610-45b0-bfaa-e6068a3259dc)